src/engagement/engagement-config.ts

```diff
@@ -36,13 +36,46 @@ export const DEFAULT_ENGAGEMENT_WEIGHTS: EngagementWeights = {
   linkedPullRequests: 2
 }
 
+/**
+ * Safely resolves and validates a config file path to prevent path traversal attacks
+ * @param basePath The base workspace path
+ * @param relativePath The relative path to the config file
+ * @returns The resolved and validated absolute path
+ */
+function safeConfigPath(basePath: string, relativePath: string): string {
```

Comment:
All plaths should be relative to the workspace root. Move this into a `src/utils.ts` and rename the function to `safePath`. Please also make sure all path loading from the user is using this.